sorter.py

```diff
@@ -0,0 +1,73 @@
+from order import Order
+
+
+class Sorter:
+    def __init__(self, permutation, sorted_sequence):
+        """
+        Order is a class implementing comparison and min in our ordering.
+        :param list(int) self.permutation: Initial self.permutation to sort
+        :param list(int) sorted_sequence: Order we want to achieve with self.permutation
+        """
+        self.N = len(sorted_sequence)
+        self.order = Order(sorted_sequence)
+        self.sorted_sequence = sorted_sequence
+        self.permutation = permutation
+
+    def _swap_values(self, min_value_i, value_i):
+        """
+        Swapping value with min value using 0 as buffer
+        :param int min_value_i: Index of min value to swap
+        :param int value_i: Index of value to swap
+        """
+        if min_value_i == value_i:
+            return
+
+        zero_i = self.permutation.index(0)
+
+        if value_i != zero_i:  # we don't need to swap 0 with 0
+            self.permutation[value_i], self.permutation[zero_i] = self.permutation[zero_i], self.permutation[value_i]
+            print(self.permutation)
+            value_i, zero_i = zero_i, value_i
+
+        self.permutation[min_value_i], self.permutation[zero_i] = self.permutation[zero_i], self.permutation[min_value_i]
```

Comment:
Better would be to make the lines shorter, here e.g. by making = separately for 2 variables?